### ISSUE TITLE
Bug: slotscheck pre-commit misconfiguration

### ISSUE THREAD

**Issue by ariebovenberg** [litestar-org/litestar#3251]
### Description

👋 Hi there @Goldziher. I'm happy to see you're still using slotscheck, but I noticed that the pre-commit `exclude` option is configured so that it isn't run on most files:

```
$ pre-commit run slotscheck --all-files -v
All OK!
stats:
  modules:     8
    checked:   8
    excluded:  0
    skipped:   5

  classes:     1
    has slots: 1
    no slots:  0
    n/a:       0
```

8 modules scanned is way too few, so I had a look.  It took me a while to realize that the pre-commit option `exclude: "test_*|docs|.github"`  regex matches `test` and `litestar` contains the string `test` 😬.

It also looks like the `flake8-dunder-all` pre-commit would have the same issue 👀. Might be worth scanning any other tooling you have set up to ignore things with "test" in the name 😄 

I'll have a look now and, if the change is small, submit a PR shortly.

### Litestar Version

main

### Platform

- [ ] Linux
- [ ] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/3251">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/3251/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/3251/pledge.svg">
</picture>
</a>

**Comment by peterschutt** [comment]
> litestar contains the string test 😬.

:D oh dear

**Comment by ariebovenberg** [comment]
This has been fixed in #3275, closed.
